### PR TITLE
[DOCS][ESQL] Add object type to unsupported field types list

### PR DESCRIPTION
**Pull request by leemthompo**

### CODE REVIEW

File: docs/reference/esql/esql-limitations.asciidoc [hunk 1]
```diff
@@ -64,6 +64,7 @@ include::processing-commands/limit.asciidoc[tag=limitation]
 ** `ip_range`
 ** `long_range`
 ** `nested`
+** `object`
```
Comment (nik9000) [inline docs/reference/esql/esql-limitations.asciidoc:67]: The story here is complicated. Like, what does "support" mean here. We load fields *in* an `object` field just fine. But we don't support loading the entire `object` field as a tree structure.

Comment (nik9000) [reply docs/reference/esql/esql-limitations.asciidoc:67]: I guess move it.... I *think* it'd look funky to have a bunch of nuance in this list. Especially around the other fields where the answer is pretty simple - we just can't touch these fields.

Comment (leemthompo) [reply docs/reference/esql/esql-limitations.asciidoc:67]: Yeah I guess the question is what weirdnesses are folks going to hit if they try to work with `object` fields in ES|QL, and are they worth enumerating here at this point in time?

Comment (nik9000) [reply docs/reference/esql/esql-limitations.asciidoc:67]: It's worth double checking. They don't come back in `*`. If you refer to them directly they are unsupported, but I don't know what that'll look like. But that's not a "normal" thing to do.

**Comment by leemthompo** [comment]
Let's park this and see if it becomes a recurring question, rather than writing a load of caveat-filled verbiage

**Pull request closed without merging**